deployer/README.md

```diff
@@ -131,6 +131,36 @@ This is to make it convenient in GitHub Actions to control the
 execution purely based on the presence of the
 environment variable.
 
+### About Elasticsearch aliases
+
+The default behavior is that each day you get a different index name.
+E.g. `mdn_docs_20210331`. And then there's an alias with a more "generic" name.
+E.g. `mdn_docs`. It's the alias name that Kuma uses to send search queries to.
+
+The way indexing works is that we leave the existing index and its alias in place,
+then we fill up a new index and once that worked, we "move the alias" and
+delete the old index. To demonstrate, consider this example timeline:
+
+- Yesterday: index `mdn_docs_20210330` and `mdn_docs --> mdn_docs_20210330`
+- Today:
+  - create index `mdn_docs_20210331`
+  - populate `mdn_docs_20210331` (could take a long time)
+  - remove alias `mdn_docs`
+  - put new alias `mdn_docs --> mdn_docs_20210331`
```

Comment:
No, I don't think they're a "single atomic operation". If, god forbid, the power was to go out between the `remove_alias` and the `put_alias` we'd be screwed. But it's the only way I could figure it out to work. 

I actually tested this. If I commented out the `remove_alias()` step what would happen is that `mdn_docs --> mdn_docs_YESTERDAY AND mdn_docs_TODAY` and when you make a search through Kuma you get twice as many matches. 

What would be ideal is to use some sort of atomic operation like you can do with a  proper relational database. 

Another alternative would be to rearrange the order so it's 
```
  - put new alias `mdn_docs --> mdn_docs_TODAY`
  - remove alias `mdn_docs --> mdn_docs_YESTERDAY`
 ```
then, if the power was to go out in between those two operations you'd just have duplicates of every search result. 

Perhaps I'm wrong! Perhaps there is a fancy Elasticsearch operation where you bundle them into 1 single POST request. Want me to look deeper into that? I didn't see anything in the Python API but it looks like mayhaps it's possible with the raw JSON API.

Reply:
Oh, I see, but I think the underlying `Elasticsearch` client that `elasticsearch_dsl` uses supports it via `client.indices.update_aliases()` (see https://elasticsearch-py.readthedocs.io/en/v7.12.0/api.html#elasticsearch.client.IndicesClient.update_aliases and https://www.elastic.co/guide/en/elasticsearch/reference/7.12/indices-aliases.html#indices-aliases-api-rename-alias-ex) but it doesn't seems that `elasticsearch_dsl` exposes that?

Reply:
Hm, I wonder if you pass the `body` keyword argument to `put_aliases()` (using the `remove` and `add` `actions`) it would work? It seems so strange that this isn't easier to do using `elasticsearch_dsl` as it seems that it would be a super common need.

Reply:
Hm, I'm reading the `elasticsearch_dsl` source code, and it might work to use:
```python
document_index.connection.update_aliases(body)
```
where I guess `body` is the underlying JSON body as described in https://www.elastic.co/guide/en/elasticsearch/reference/7.12/indices-aliases.html#indices-aliases-api-request-body ?

Reply:
I think we should look some more into doing this atomically. From my reading of the ElasticSearch API, it's possible. In addition to what I found in my comments above, see the section `Reindexing with no downtime` in https://www.objectrocket.com/blog/elasticsearch/elasticsearch-aliases/.